core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -305,4 +334,76 @@ public void start() {
         }
         logger().info("Docker compose has finished running");
     }
-}
\ No newline at end of file
+}
+
+class LocalDockerCompose implements DockerCompose {
+    private final File composeFile;
+    private final String identifier;
+    private String cmd = "";
+    private Map<String, String> env = new HashMap<>();
+
+    public LocalDockerCompose(File composeFile, String identifier) {
+        this.composeFile = composeFile;
+        this.identifier = identifier;
+    }
+
+    @Override
+    public DockerCompose withCommand(String cmd) {
+        this.cmd = cmd;
+        return this;
+    }
+
+    @Override
+    public DockerCompose withEnv(Map<String, String> env) {
+        this.env = env;
+        return this;
+    }
+
+    @Override
+    public void start() {
+        try {
+            final ProcessBuilder pb = new ProcessBuilder(Splitter.onPattern(" ")
+                    .omitEmptyStrings()
+                    .splitToList("docker-compose " + cmd));
+
+            pb.redirectErrorStream(true);
+
+            final Map<String, String> environment = pb.environment();
+            environment.putAll(env);
+            environment.put("COMPOSE_PROJECT_NAME", identifier);
+
+            final File pwd = composeFile.getAbsoluteFile().getParentFile().getAbsoluteFile();
+            environment.put("COMPOSE_FILE", new File(pwd, composeFile.getAbsoluteFile().getName()).getAbsolutePath());
+
+            logger().info("Local Docker compose is running command: {}", cmd);
+            pb.directory(pwd);
+            final Process p = pb.start();
+
+            // wait for the compose process to stop, which should only happen after it has spawned all the service containers
+            p.waitFor(120, TimeUnit.SECONDS);
```

Comment:
Should this timeout be longer, especially to allow for pulling/building of containers..? 
I'm not sure there's actually any easy timeout length, so maybe just let it be infinite?